### PR TITLE
Chore: Upgrades Cypress to 8.4.1

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Upgrades Cypress and peer libs.

https://docs.cypress.io/guides/references/changelog#8-0-0

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: e2e/run-suite [hunk 1]
```diff
@@ -28,5 +28,5 @@ fi
 cd packages/grafana-e2e
 
 yarn $CMD --env BASE_URL=$URL,SLOWMO=$SLOWMO \
-  --config defaultCommandTimeout=30000,integrationFolder=../../e2e/$SUITE/specs,screenshotsFolder=../../e2e/$SUITE/screenshots,videosFolder=../../e2e/$SUITE/videos,fileServerFolder=./cypress,viewportWidth=1920,viewportHeight=1080,trashAssetsBeforeRuns=false \
+  --config defaultCommandTimeout=30000,integrationFolder=../../e2e/$SUITE/specs,screenshotsFolder=../../e2e/$SUITE/screenshots,videosFolder=../../e2e/$SUITE/videos,fileServerFolder=./cypress,viewportWidth=1920,viewportHeight=1080,trashAssetsBeforeRuns=false,videoUploadOnPasses=false \
```
Comment (hugohaggmark) [inline e2e/run-suite:31]: My thinking is that `videoUploadOnPasses=false` might speed up passing e2e tests a bit, but might just be me overthinking it 🤔

File: e2e/run-suite [hunk 2]
```diff
@@ -17,7 +17,7 @@ SUITE=${SUITE:-$DEFAULT_SUITE}
 if [ "$1" == "debug" ]; then
     echo -e "Debug mode"
     SLOWMO=1
-    PARAMS="--headed --no-exit"
```
Comment (hugohaggmark) [inline e2e/run-suite:20]: No need for `headed` as this is the default behavior now.

**Comment by hugohaggmark** [comment]
Some prel. results:

**Before**
image
image
image
image
image

**After**
image

Maybe `videoUploadOnPasses=false` setting has effect on the time 🤔

File: e2e/run-suite (hunk 1 above)
Comment (vickyyyyyyy) [reply e2e/run-suite:31]: Having a look at the docs: https://docs.cypress.io/guides/references/configuration#Videos and they mention
```
This only applies when recording your runs to the Dashboard. 
```
I'm not too familiar with the Cypress dashboard, but it sounds like this won't impact our build or local runs? Not sure though 🤔

Comment (hugohaggmark) [reply e2e/run-suite:31]: @vickyyyyyyy bad docs, it seems like it has effects on the time see https://github.com/grafana/grafana/pull/39573#issuecomment-925737787

**Review by dprokop — APPROVED** [review]
🐎

File: packages/grafana-e2e/cypress/plugins/typescriptPreprocessor.js [hunk 3]
```diff
@@ -22,6 +22,9 @@ const webpackOptions = {
         use: [
           {
             loader: 'ts-loader',
+            options: {
+              transpileOnly: true,
```
Comment (zoltanbedi) [inline packages/grafana-e2e/cypress/plugins/typescriptPreprocessor.js:26]: I'd say that the boost comes from here.

Comment (hugohaggmark) [reply packages/grafana-e2e/cypress/plugins/typescriptPreprocessor.js:26]: Nope, it was `06:51` before that change

**Pull request merged**